### ISSUE TITLE
Failed Test: yarn test [v1.9.0]

### ISSUE THREAD

**Issue by Jovonni** [graphprotocol/contracts#533]
```bash
$ yarn test
yarn run v1.22.17
$ scripts/test
$ SKIP_LOAD=true scripts/build
$ hardhat compile
Compiling 77 files with 0.7.6
@openzeppelin/contracts/token/ERC20/ERC20.sol:55:5: Warning: Visibility for constructor is ignored. If you want the contract to be non-deployable, making it "abstract" is sufficient.
    constructor (string memory name_, string memory symbol_) public {
    ^ (Relevant source part starts here and spans across multiple lines).

contracts/upgrades/GraphProxy.sol:69:61: Warning: Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.
    function admin() external ifAdminOrPendingImpl returns (address) {
                                                            ^-----^

contracts/upgrades/GraphProxy.sol:82:70: Warning: Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.
    function implementation() external ifAdminOrPendingImpl returns (address) {
                                                                     ^-----^

contracts/upgrades/GraphProxy.sol:95:77: Warning: Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.
    function pendingImplementation() external ifAdminOrPendingImpl returns (address) {
                                                                            ^-----^

contracts/tests/ens/IENS.sol: Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.

contracts/tests/ens/IENS.sol:11:9: Warning: This declaration has the same name as another declaration.
        address owner,
        ^-----------^
contracts/tests/ens/IENS.sol:5:5: The other declaration is here:
    function owner(bytes32 node) external view returns (address);
    ^-----------------------------------------------------------^

contracts/tests/ens/IPublicResolver.sol: Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.

contracts/tests/ens/ITestRegistrar.sol: Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.

contracts/tests/RewardsManagerMock.sol: Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.

Generating typings for: 78 artifacts in dir: build/types for target: ethers-v5
Successfully generated 129 typings!
Compilation finished successfully
No need to generate any newer typings.


  Curation:Config
    defaultReserveRatio
      ✓ should set `defaultReserveRatio` (231ms)
      ✓ reject set `defaultReserveRatio` if out of bounds (71ms)
      ✓ reject set `defaultReserveRatio` if not allowed
    minimumCurationDeposit
      ✓ should set `minimumCurationDeposit` (210ms)
      ✓ reject set `minimumCurationDeposit` if out of bounds
      ✓ reject set `minimumCurationDeposit` if not allowed
    curationTaxPercentage
      ✓ should set `curationTaxPercentage` (281ms)
      ✓ reject set `curationTaxPercentage` if out of bounds
      ✓ reject set `curationTaxPercentage` if not allowed
    curationTokenMaster
      ✓ should set `curationTokenMaster` (137ms)
      ✓ reject set `curationTokenMaster` to empty value
      ✓ reject set `curationTokenMaster` to non-contract
      ✓ reject set `curationTokenMaster` if not allowed

  Curation
    bonding curve
      ✓ reject convert signal to tokens if subgraph deployment not initted
      ✓ convert signal to tokens (1061ms)
      ✓ convert signal to tokens (with curation tax) (1273ms)
      ✓ convert tokens to signal
      ✓ convert tokens to signal if non-curated subgraph
    curate
      ✓ reject deposit below minimum tokens required (44ms)
      ✓ should deposit on a subgraph deployment (618ms)
      ✓ should get signal according to bonding curve (972ms)
      ✓ should get signal according to bonding curve (and account for curation tax) (1182ms)
      ✓ should revert curate if over slippage (45ms)
    collect
      > not curated
        ✓ reject collect tokens distributed to the curation pool (279ms)
      > curated
        ✓ reject collect tokens distributed from invalid address
        ✓ should collect tokens distributed to the curation pool (1026ms)
        ✓ should collect tokens and then unsignal all (1049ms)
        ✓ should collect tokens and then unsignal multiple times (1784ms)
    burn
      ✓ reject redeem more than a curator owns
      ✓ reject redeem zero signal
      ✓ should allow to redeem *partially* (605ms)
      ✓ should allow to redeem *fully* (571ms)
      ✓ should allow to redeem back below minimum deposit (2075ms)
      ✓ should revert redeem if over slippage
      ✓ should not re-deploy the curation token when signal is reset (1269ms)
    conservation
      ✓ should match multiple deposits and redeems back to initial state (13178ms)
    multiple minting
      ✓ should mint less signal every time due to the bonding curve (4502ms)
      ✓ should mint when using the edge case of linear function (4018ms)

  DisputeManager:Config
    configuration
      arbitrator
        ✓ should set `arbitrator` (119ms)
        ✓ reject set `arbitrator` if not allowed
        ✓ reject set `arbitrator` to address zero
      minimumDeposit
        ✓ should set `minimumDeposit` (116ms)
        ✓ reject set `minimumDeposit` if not allowed
      fishermanRewardPercentage
        ✓ should set `fishermanRewardPercentage` (218ms)
        ✓ reject set `fishermanRewardPercentage` if out of bounds
        ✓ reject set `fishermanRewardPercentage` if not allowed
      slashingPercentage
        ✓ should set `slashingPercentage` (232ms)
        ✓ reject set `slashingPercentage` if out of bounds
        ✓ reject set `slashingPercentage` if not allowed

  DisputeManager:POI
    disputes
      ✓ reject create a dispute if allocation does not exist
      ✓ reject create a dispute if indexer below stake (2219ms)
      > when indexer is staked
        ✓ should create a dispute (307ms)
        > when dispute is created
          ✓ reject create duplicated dispute
          accept a dispute
            ✓ should resolve dispute, slash indexer and reward the fisherman (424ms)

  DisputeManager:Query
    disputes
      ✓ reject create a dispute if attestation does not refer to valid indexer (39ms)
      ✓ reject create a dispute if indexer below stake (2116ms)
      > when indexer is staked
        create dispute
          ✓ reject fisherman deposit below minimum required
          ✓ should create a dispute (375ms)
        accept a dispute
          ✓ reject to accept a non-existing dispute
        reject a dispute
          ✓ reject to reject a non-existing dispute
        draw a dispute
          ✓ reject to draw a non-existing dispute
        > when dispute is created
          create a dispute
            ✓ should create dispute if receipt is equal but for other indexer (372ms)
            ✓ should create dispute as long as it is from different fisherman (362ms)
            ✓ reject create duplicated dispute
          accept a dispute
            ✓ reject to accept a dispute if not the arbitrator
            ✓ reject to accept a dispute if not slasher (125ms)
            ✓ reject to accept a dispute if zero tokens to slash (181ms)
            ✓ should resolve dispute, slash indexer and reward the fisherman (415ms)
          reject a dispute
            ✓ reject to reject a dispute if not the arbitrator
            ✓ should reject a dispute and burn deposit (149ms)
          draw a dispute
            ✓ reject to draw a dispute if not the arbitrator
            ✓ should draw a dispute and return deposit (152ms)
    disputes for conflicting attestations
      ✓ reject if attestations are not in conflict
      ✓ should create dispute (689ms)
      ✓ should accept one dispute and resolve the related dispute (1093ms)
      ✓ should not allow to reject, user need to accept the related dispute ID to reject it (725ms)
      ✓ should draw one dispute and resolve the related dispute (811ms)

  EpochManager
    configuration
      ✓ should set `epochLength (122ms)
      ✓ reject set `epochLength` if zero
    epoch lifecycle
      calculations
        ✓ should return correct block number
        ✓ should return same starting block if we stay on the same epoch
        ✓ should return next starting block if we move to the next epoch
        ✓ should return next epoch if advance > epochLength
      progression
        > epoch not run
          ✓ should return that current epoch is not run
          ✓ should run new epoch (81ms)
        > epoch run
          ✓ should return current epoch is already run
          ✓ reject run new epoch

  GNS
    Publishing names and versions
      setDefaultName
        ✓ setDefaultName emits the event (109ms)
        ✓ setDefaultName fails if not owner
      updateSubgraphMetadata
        ✓ updateSubgraphMetadata emits the event (106ms)
        ✓ updateSubgraphMetadata fails if not owner
      isPublished
        ✓ should return if the subgraph is published (195ms)
      publishNewSubgraph
        ✓ should publish a new subgraph and first version with it (195ms)
        ✓ should publish a new subgraph with an incremented value (359ms)
        ✓ should reject publish if not sent from owner
        ✓ should prevent subgraphDeploymentID of 0 to be used
      publishNewVersion
        ✓ should publish a new version on an existing subgraph (1868ms)
        ✓ should reject a new version with the same subgraph deployment ID
        ✓ should reject publishing a version to a numbered subgraph that does not exist
        ✓ reject if not the owner
        ✓ should fail when upgrade tries to point to a pre-curated (909ms)
        ✓ should fail when trying to upgrade when there is no nSignal (741ms)
        ✓ should fail when subgraph is deprecated (742ms)
      deprecateSubgraph
        ✓ should deprecate a subgraph (693ms)
        ✓ should prevent a deprecated subgraph from being republished (656ms)
        ✓ reject if the subgraph does not exist
        ✓ reject deprecate if not the owner
    Curating on names
      mintNSignal()
        ✓ should deposit into the name signal curve (1432ms)
        ✓ should fail when name signal is disabled (413ms)
        ✓ should fail if you try to deposit on a non existing name
        ✓ reject minting if under slippage (329ms)
      burnNSignal()
        ✓ should withdraw from the name signal curve (854ms)
        ✓ should fail when name signal is disabled (662ms)
        ✓ should fail when the curator tries to withdraw more nSignal than they have
        ✓ reject burning if under slippage (51ms)
      withdraw()
        ✓ should withdraw GRT from a disabled name signal (796ms)
        ✓ should fail if not disabled
        ✓ should fail when there is no more GRT to withdraw (922ms)
        ✓ should fail if the curator has no nSignal (640ms)
      multiple minting
        ✓ should mint less signal every time due to the bonding curve (6953ms)
        ✓ should mint when using the edge case of linear function (6708ms)
        setOwnerTaxPercentage
          ✓ should set `ownerTaxPercentage` (99ms)
          ✓ reject set `ownerTaxPercentage` if out of bounds
          ✓ reject set `ownerTaxPercentage` if not allowed
    Two named subgraphs point to the same subgraph deployment ID
      ✓ handle initialization under minimum signal values (2233ms)
    batch calls
      ✓ should publish new subgraph and mint signal in single transaction (1327ms)
      ✓ should revert if batching a call to non-authorized function
      ✓ should revert if batching a call to initialize
      ✓ should revert if trying to call a private function

  Managed
    setContractProxy()
      ✓ should set contract proxy and test get contract proxy
      ✓ reject set contract proxy to address zero
      ✓ reject set contract proxy if not governor
    unsetContractProxy()
      ✓ should unset contract proxy (85ms)
      ✓ reject to call if not authorized caller
    updateController()
      ✓ should update controller on a manager (99ms)
      ✓ should fail updating controller when not called by governor
      ✓ reject update controller to address zero
    setController()
      ✓ should fail setting controller when not called from Controller
    setPauseGuardian()
      ✓ should set the pause guardian
      ✓ reject to call if not authorized caller
      ✓ reject set pause guardian to address zero

  GraphGovernance
    proposals
      ✓ should create a proposal (90ms)
      ✓ reject create a proposal if not allowed
      ✓ reject create a proposal with empty proposalId
      ✓ reject create a proposal with empty votes proof
      ✓ reject create a proposal with empty resolution
      > proposal created
        ✓ should update a proposal (83ms)
        ✓ reject create a duplicated proposal
        ✓ reject update a non-existing proposal
        ✓ reject update with empty proposalId
        ✓ reject update a proposal with empty votes proof
        ✓ reject update a proposal with empty resolution

  Governed
    ✓ should reject transfer if not allowed
    ✓ should transfer and accept (43ms)

  Pausing
    ✓ should set pause guardian
    ✓ should fail pause guardian when not governor
    ✓ should set partialPaused and unset from governor and guardian (197ms)
    ✓ should fail partial pause if not guardian or governor
    ✓ should check that a function fails when partialPause is set (49ms)
    ✓ should set pause and unset from governor and guardian (193ms)
    ✓ should fail pause if not guardian or governor
    ✓ should check that a function fails when pause is set (60ms)

  GraphToken
    permit
      ✓ should permit max token allowance (86ms)
      ✓ should permit max token allowance (77ms)
      ✓ reject to transfer more tokens than approved by permit (136ms)
      ✓ reject use two permits with same nonce (64ms)
      ✓ reject use expired permit
      ✓ reject permit if holder address does not match
      ✓ should deny transfer from if permit was denied (161ms)
    mint
      addMinter
        ✓ reject add a new minter if not allowed
        ✓ should add a new minter
      mint
        ✓ reject mint if not minter
      > when is minter
        mint
          ✓ should mint
          ✓ should mint if governor
        removeMinter
          ✓ reject remove a minter if not allowed
          ✓ should remove a minter (60ms)
        renounceMinter
          ✓ should renounce to be a minter (58ms)

  AllocationExchange
    config
      ✓ should set an authority (88ms)
      ✓ reject set an authority if not allowed
      ✓ reject set an empty authority
      ✓ should allow to approve all tokens to staking contract
    withdraw funds
      ✓ should withdraw to destination (50ms)
      ✓ reject withdraw zero amount
      ✓ reject withdraw to zero destination
      ✓ reject withdraw if not allowed
    redeem vouchers
      ✓ redeem a voucher (952ms)
      ✓ reject double spending of a voucher (946ms)
      ✓ reject redeem voucher for invalid allocation (113ms)
      ✓ reject redeem voucher not signed by the authority (41ms)
      ✓ reject redeem voucher with empty amount
      ✓ reject redeem voucher with invalid signature

  WithdrawHelper
    execute withdrawal
      ✓ withdraw tokens from the CMC to staking contract through WithdrawHelper (998ms)
      ✓ withdraw tokens from the CMC to staking contract through WithdrawHelper (invalid allocation) (273ms)
      ✓ reject collect data with no staking address
      ✓ reject collect data with no allocation

  Rewards
    configuration
      issuance rate update
        ✓ reject set issuance rate if unauthorized
        ✓ reject set issuance rate to less than minimum allowed
        ✓ should set issuance rate to minimum allowed (127ms)
        ✓ should set issuance rate (137ms)
      subgraph availability service
        ✓ reject set subgraph oracle if unauthorized
        ✓ should set subgraph oracle if governor (92ms)
        ✓ reject to deny subgraph if not the oracle
        ✓ should deny subgraph (169ms)
        ✓ reject deny subgraph w/ many if not the oracle
        ✓ should deny subgraph w/ many (184ms)
    issuing rewards
      getNewRewardsPerSignal
        ✓ accrued per signal when no tokens signalled (180ms)
        ✓ accrued per signal when tokens signalled (823ms)
        ✓ accrued per signal when signalled tokens w/ many subgraphs (1522ms)
      updateAccRewardsPerSignal
        ✓ update the accumulated rewards per signal state (1696ms)
        ✓ update the accumulated rewards per signal state after many blocks (958ms)
      getAccRewardsForSubgraph
        ✓ accrued for each subgraph (1662ms)
      onSubgraphSignalUpdate
        ✓ update the accumulated rewards for subgraph state (1003ms)
      getAccRewardsPerAllocatedToken
        ✓ accrued per allocated token (1548ms)
      onSubgraphAllocationUpdate
        ✓ update the accumulated rewards for allocated tokens state (1711ms)
      getRewards
        ✓ calculate rewards using the subgraph signalled + allocated tokens (1520ms)
      takeRewards
        ✓ should distribute rewards on closed allocation and stake (2267ms)
        ✓ should distribute rewards on closed allocation and send to destination (2470ms)
        ✓ should distribute rewards on closed allocation w/delegators (2594ms)
        ✓ should deny rewards if subgraph on denylist (2277ms)
    pow
	r: 1.0 => c: 1.0
	r: 1.000000074262046768 => c: 1.004286655269483037
	r: 1.000000148524093536 => c: 1.008591685631969614
	r: 1.000000222786140304 => c: 1.012915169852438674
	r: 1.000000297048187072 => c: 1.017257187033415327
	r: 1.00000037131023384 => c: 1.021617816616507562
	r: 1.000000445572280608 => c: 1.025997138383963331
	r: 1.000000519834327376 => c: 1.030395232459812669
	r: 1.000000594096374144 => c: 1.034812179311705175
	r: 1.000000668358420912 => c: 1.039248059752073802
	r: 1.00000074262046768 => c: 1.043702954939860526
	r: 1.000000816882514448 => c: 1.048176946381841108
	r: 1.000000891144561216 => c: 1.052670115934159722
      ✓ exponentiation works under normal boundaries (annual rate from 1% to 700%, 90 days period) (94ms)
    edge scenarios
      ✓ close allocation on a subgraph that no longer have signal (2699ms)

  ServiceRegistry
    register
      ✓ should allow registering (108ms)
      ✓ should allow registering with a very long string (123ms)
      ✓ should allow updating a registration (215ms)
      ✓ reject registering empty URL
      operator
        ✓ reject register from unauthorized operator
        ✓ should register from operator (251ms)
    unregister
      ✓ should unregister existing registration (193ms)
      ✓ reject unregister non-existing registration
      operator
        ✓ reject unregister from unauthorized operator (127ms)
        ✓ should unregister from operator (316ms)

  Staking:Allocation
    operators
      ✓ should set operator (108ms)
      ✓ should unset operator (181ms)
    rewardsDestination
      ✓ should set rewards destination (179ms)
    allocate
      ✓ reject allocate zero tokens
      ✓ reject allocate with invalid allocationID
      ✓ reject allocate if no tokens staked
      > when staked
        ✓ reject allocate more than available tokens
        ✓ should allocate (561ms)
        ✓ should allocate on behalf of indexer (826ms)
        ✓ reject allocate reusing an allocation ID (1038ms)
        reject allocate on invalid proof
          ✓ invalid message
          ✓ invalid proof signature format
    collect
      ✓ reject collect if invalid collection
      ✓ reject collect if allocation does not exist
      ✓ should collect funds from asset holder (1318ms)
      ✓ should collect funds from asset holder and distribute curation fees (1261ms)
      ✓ should collect funds from asset holder + protocol fee + curation fees (1385ms)
      ✓ should collect zero tokens (355ms)
      ✓ should collect from a settling allocation but reject after dispute period (5057ms)
    closeAllocation
      ✓ reject close a non-existing allocation
      ✓ reject close before at least one epoch has passed (63ms)
      ✓ reject close if not the owner of allocation (472ms)
      ✓ reject close if allocation is already closed (1436ms)
      ✓ should close an allocation (2141ms)
      ✓ should close an allocation (by operator) (2465ms)
      ✓ should close an allocation (by delegator) (3975ms)
      ✓ should close many allocations in batch (5564ms)
    closeAndAllocate
      ✓ should close and create a new allocation (2664ms)
    claim
      ✓ reject claim for non-existing allocation (40ms)
      ✓ reject claim if allocation is not closed (53ms)
      > when allocation closed
        ✓ reject claim if closed but channel dispute epochs has not passed (42ms)
        ✓ should claim rebate (1623ms)
        ✓ should claim rebate with restake (1665ms)
        ✓ should claim rebate (by public) (1063ms)
        ✓ should claim rebate (by operator) (1390ms)
        ✓ should claim many rebates with restake (970ms)
        ✓ reject claim if already claimed (1441ms)

  Staking:Config
    minimumIndexerStake
      ✓ should set `minimumIndexerStake` (223ms)
      ✓ reject set `minimumIndexerStake` if not allowed
      ✓ reject set `minimumIndexerStake` to zero
    setSlasher
      ✓ should set `slasher` (452ms)
      ✓ reject set `slasher` if not allowed
      ✓ reject set `slasher` for zero
    setAssetHolder
      ✓ should set `assetHolder` (692ms)
      ✓ reject set `assetHolder` if not allowed (39ms)
      ✓ reject set `assetHolder` to address zero
    channelDisputeEpochs
      ✓ should set `channelDisputeEpochs` (213ms)
      ✓ reject set `channelDisputeEpochs` if not allowed
      ✓ reject set `channelDisputeEpochs` to zero
    curationPercentage
      ✓ should set `curationPercentage` (180ms)
      ✓ reject set `curationPercentage` if out of bounds (45ms)
      ✓ reject set `curationPercentage` if not allowed
    protocolPercentage
      ✓ should set `protocolPercentage` (773ms)
      ✓ reject set `protocolPercentage` if out of bounds
      ✓ reject set `protocolPercentage` if not allowed
    maxAllocationEpochs
      ✓ should set `maxAllocationEpochs` (198ms)
      ✓ reject set `maxAllocationEpochs` if not allowed
    thawingPeriod
      ✓ should set `thawingPeriod` (214ms)
      ✓ reject set `thawingPeriod` if not allowed (42ms)
      ✓ reject set `thawingPeriod` to zero
    rebateRatio
      ✓ should be setup on init
      ✓ should set `rebateRatio` (222ms)
      ✓ reject set `rebateRatio` if out of bounds (57ms)
      ✓ reject set `rebateRatio` if not allowed

  Staking::Delegation
    configuration
      delegationRatio
        ✓ should set `delegationRatio` (225ms)
        ✓ reject set `delegationRatio` if not allowed
      delegationParametersCooldown
        ✓ should set `delegationParametersCooldown` (216ms)
        ✓ reject set `delegationParametersCooldown` if not allowed (48ms)
      delegationTaxPercentage
        ✓ should set `delegationTaxPercentage` (664ms)
        ✓ reject set `delegationTaxPercentage` if out of bounds
        ✓ reject set `delegationTaxPercentage` if not allowed
      delegationParameters
        ✓ reject to set if under cooldown period (209ms)
        ✓ reject to set if cooldown below the global configuration (285ms)
        ✓ reject to set parameters out of bound (39ms)
        ✓ should set parameters (204ms)
        ✓ should init delegation parameters on first stake (431ms)
        ✓ should init delegation parameters on first stake using stakeTo() (493ms)
    lifecycle
      delegate
        ✓ reject delegate with zero tokens
        ✓ reject delegate to empty address (38ms)
        ✓ reject delegate to non-staked indexer (80ms)
        ✓ should delegate tokens and account shares proportionally (3627ms)
        ✓ should delegate a high amount of tokens (1286ms)
        delegation tax
          ✓ should delegate and burn delegation deposit tax (0.0001%) (967ms)
          ✓ should delegate and burn delegation deposit tax (1%) (686ms)
          ✓ reject delegate with delegation deposit tax (100%) (300ms)
      undelegate
        ✓ reject to undelegate zero shares
        ✓ reject to undelegate more shares than owned
        ✓ should exchange delegation pool shares for tokens (1480ms)
        ✓ should undelegate properly when multiple delegations (6110ms)
        1) should undelegate and withdraw freed tokens from unbonding period
      withdraw
        ✓ reject withdraw if no funds available (45ms)
        ✓ reject withdraw before unbonding period (1284ms)
        ✓ should withdraw after waiting an unbonding period (2746ms)
        ✓ should withdraw after waiting an unbonding period (with redelegation) (3134ms)
    use of delegated funds
      ✓ revert allocate when capacity is not enough (863ms)
      ✓ should allocate using full delegation capacity (1747ms)
      ✓ should account delegation for indexer capacity properly (1011ms)
      ✓ should send delegation cut of query fees to delegation pool (5353ms)
      ✓ revert if it cannot assign the smallest amount of shares (5124ms)

  Staking:Rebate
    should match cobb-douglas Solidity implementation
      normal test case
        ✓ alpha 0.90 (113ms)
        ✓ alpha 0.25 (122ms)
        ✓ alpha 0.33~ (72ms)
        ✓ alpha 0.005 (97ms)
        ✓ alpha 1 (175ms)
      edge #1 test case
        ✓ alpha 0.90 (39ms)
        ✓ alpha 0.25
        ✓ alpha 0.33~
        ✓ alpha 0.005
        ✓ alpha 1
    should match rewards out from rebates
      normal test case
        ✓ alpha 0.90 (888ms)
        ✓ alpha 0.25 (800ms)
        ✓ alpha 0.33~ (832ms)
        ✓ alpha 0.005 (806ms)
        ✓ alpha 1 (804ms)
      edge #1 test case
        ✓ alpha 0.90 (794ms)
        ✓ alpha 0.25 (905ms)
        ✓ alpha 0.33~ (770ms)
        ✓ alpha 0.005 (720ms)
        ✓ alpha 1 (714ms)
    should always be that sum of rebate rewards obtained <= to total rewards
      normal test case
        ✓ alpha 0.90 (845ms)
        ✓ alpha 0.25 (796ms)
        ✓ alpha 0.33~ (697ms)
        ✓ alpha 0.005 (955ms)
        ✓ alpha 1 (738ms)
      edge #1 test case
        ✓ alpha 0.90 (546ms)
        ✓ alpha 0.25 (455ms)
        ✓ alpha 0.33~ (512ms)
        ✓ alpha 0.005 (708ms)
        ✓ alpha 1 (816ms)

  Staking:Stakes
    > when not staked
      hasStake
        ✓ should not have stakes
      stake
        ✓ reject stake zero tokens
        ✓ reject stake less than minimum indexer stake (44ms)
        ✓ should stake tokens (411ms)
        ✓ should stake tokens = minimumIndexerStake (444ms)
      unstake
        ✓ reject unstake tokens
      slash
        ✓ reject slash indexer
    > when staked
      hasStake
        ✓ should have stakes
      stake
        ✓ should allow re-staking (403ms)
        ✓ reject to stake under the minimum indexer stake after unstake + slash (768ms)
      unstake
        ✓ should unstake and lock tokens for thawing period (314ms)
        ✓ should unstake and lock tokens for (weighted avg) thawing period if repeated (756ms)
        ✓ should unstake and withdraw if some tokens are unthawed (1024ms)
        ✓ should unstake available tokens even if passed a higher amount (303ms)
        ✓ reject unstake zero tokens
        ✓ reject unstake under the minimum indexer stake (79ms)
        ✓ reject unstake under the minimum indexer stake w/multiple unstake (461ms)
        ✓ should allow unstake of full amount (326ms)
        ✓ should allow unstake of full amount with no upper limits (1103ms)
      withdraw
        ✓ reject withdraw if no tokens available
        ✓ should withdraw if tokens available (743ms)
      slash
        ✓ should slash indexer and give reward to beneficiary slash>reward (462ms)
        ✓ should slash indexer and give reward to beneficiary slash=reward (390ms)
        ✓ should slash indexer even when overallocated (1790ms)
        ✓ reject to slash zero tokens
        ✓ reject to slash indexer if caller is not slasher
        ✓ reject to slash indexer if beneficiary is zero address
        ✓ reject to slash indexer if reward is greater than slash amount

  Gdai and Gsr
    gdai
      ✓ should set `governor`
      ✓ should allow governor to mint (69ms)
    gsr
      ✓ should set `governor`
      ✓ should set savings rate (85ms)
      ✓ should test drip() and join() (169ms)
      ✓ should test drip() and join() and exit (627ms)

  Upgrades
    GraphProxyAdmin & GraphProxy
      getters
        admin()
          ✓ should get the proxy admin of a proxy contract
          ✓ reject get admin from other than the ProxyAdmin
        implementation()
          ✓ should get implementation only from ProxyAdmin
          ✓ reject get implementation from other than the ProxyAdmin
        pendingImplementation()
          ✓ should get pending implementation only from ProxyAdmin
          ✓ reject get pending implementation from other than the ProxyAdmin
      upgrade
        ✓ should be able to upgrade contract (540ms)
        ✓ reject upgrade if not the governor of the ProxyAdmin
        ✓ reject upgrade if not using the ProxyAdmin
      acceptUpgrade
        ✓ reject accept upgrade if not using the ProxyAdmin
      acceptProxy
        ✓ reject accept proxy if not using the ProxyAdmin (232ms)
      changeProxyAdmin
        ✓ should set the proxy admin of a proxy (381ms)
        ✓ reject change admin if not the governor of the ProxyAdmin (219ms)
        ✓ reject change admin if not using the ProxyAdmin (43ms)


  407 passing (9m)
  1 failing

  1) Staking::Delegation
       lifecycle
         undelegate
           should undelegate and withdraw freed tokens from unbonding period:

      AssertionError: Expected "5" to be equal 4
      + expected - actual

       {
      -  "_hex": "0x04"
      +  "_hex": "0x05"
         "_isBigNumber": true
       }
      
      at assertArgsArraysEqual (node_modules/@ethereum-waffle/chai/dist/cjs/matchers/emit.js:58:54)
      at tryAssertArgsArraysEqual (node_modules/@ethereum-waffle/chai/dist/cjs/matchers/emit.js:65:20)
      at /Users/user_1/Documents/code/GRT/contracts/node_modules/@ethereum-waffle/chai/dist/cjs/matchers/emit.js:77:13
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at runNextTicks (internal/process/task_queues.js:64:3)
      at listOnTimeout (internal/timers.js:526:9)
      at processTimers (internal/timers.js:500:7)
      at shouldUndelegate (test/staking/delegation.test.ts:107:5)
      at Context.<anonymous> (test/staking/delegation.test.ts:472:9)



error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Judging from the following test, has the test criteria been changed:
```bash
1) Staking::Delegation
       lifecycle
         undelegate
           should undelegate and withdraw freed tokens from unbonding period:

      AssertionError: Expected "5" to be equal 4
      + expected - actual

       {
      -  "_hex": "0x04"
      +  "_hex": "0x05"
         "_isBigNumber": true
       }
```

This is on branch:
```On branch v1.9.0```

**Comment by pcarranzav** [comment]
Hi @Jovonni! Were you seeing this every time or only occasionally? If it's the latter, I think it's likely this was fixed in #554.

**Comment by tmigone** [comment]
I've verified this happens on `v1.9.0` occasionally (1 out of 3, admittedly with a small sample size). I could not however replicate the issue on `v1.11.0` or later so the problem was likely what Pablo fixed in #554. I'll close this issue, we can always get back to it if it reoccurs in the future.
